tensorflow/go/attrs_test.go

```diff
@@ -45,3 +57,47 @@ func TestOperationAttrs(t *testing.T) {
 		}
 	}
 }
+
+func TestOperationAttrsEmptyList(t *testing.T) {
+	attrs := map[string]interface{}{
+		"dtype": Float,
+	}
+
+	g := NewGraph()
+	op, err := g.AddOperation(OpSpec{
+		Type:  "Placeholder",
+		Name:  "placeholder",
+		Attrs: attrs,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	name := "foo"
+	for _, attrType := range attrTypes {
+		testListAttribute(op, name, attrType, 0, 0)
+		testListAttribute(op, name, attrType, 1, 0)
+	}
+}
+
+func TestOperationAttrsEmptyScalar(t *testing.T) {
```

Comment:
I'm not quite seeing the value of this test. For all but the `Float` type attribute, `testScalarAttribute` will return an error because the underlying C API call will return a non-OK `TF_Status`. And we're ignoring the return value, and testing it twice (lines 100, 101).

For the list valued attributes, I see we're testing a bit more (that we don't take the address of the first element of an empty slice) but we're not testing that it returns the right value and only going through the early return path in the C API.

I'm tempted to do away with these tests (and ideally replace them with ones where we actually test the various values, but I realize that will be a bit of a pain since we'll have to find operations for each attribute type).

Reply:
I see that the point here is to just test that we don't see the nil-panics that motivated this PR in the first place. So, I feel the testing here is ... incomplete.

It would be an option to use existing operations instead of adding test types, and I think coverage from that would be best.

If you insist on keeping these, I'm okay, but perhaps call the tests `Test...DoesNotPanic`, since the test itself doesn't actually validate any of the returned values.